### PR TITLE
feat: add both ID systems and validators

### LINKED ISSUE #73: Ideate on new ID system
## Idea 1: Human-Readable Prefixes

Each affiliation gets an ID with a human-readable prefix. For example, there could be a GCEP with the ID `gcep_123`. Another example would be a VCEP with the ID `vcep_456`. The part after the underscore could be an integer or it could be an alpha-numeric string (along the lines of a [UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier)). My preference would be an integer because it's easier to type in than an alpha-numeric string, and I don't think we need to worry about [enumeration attacks](https://www.upguard.com/blog/what-is-an-enumeration-attack) for the affiliations information.

### PR DESCRIPTION
**Pull request by sanchegm**
Description
- Distinguish Affiliation (10000) ID and Curation Panel ID (4000/5000) ID system in the affiliation service
- All Affiliations will have an affiliation ID, but if the type is Independent Curation Panel, curation panel ID will be null.
- Add validation to check if Independent Curation Panel type has been selected, curation panel ID is left blank. The form should not allow an Independent Curation Panel to have a curation panel ID at this time.
- Add validation to check that VCEP's curation panel ID is in the 50000 range, that GCEP's  curation panel ID is in the 40000 range, and that Affiliation IDs are in the 10000 range.

Issue Ticket Number and Link 
Related to issue #73. Should Close #73.

Checklist Remove any options that are not applicable 
[X] I have reviewed the [how-to guide](https://github.com/ClinGen/stanford-affils/pull/how-to.md) and the [contributing file](https://github.com/ClinGen/stanford-affils/CONTRIBUTING.md). 
[X] I have run required [code checks](https://github.com/ClinGen/stanford-affils/pull/how-to.md#run-code-checks) and resolved any blockers. 
[X] I have created the necessary [tests](https://github.com/ClinGen/stanford-affils/src/app_test.py) to show my changes are effective and work as intended. 
[X] I have thoroughly commented my code, especially in more complex areas. 
[X] I have updated any necessary documentation.

Screenshots / Recordings
Attempt to add new affiliation with Independent Curation Panel type and a curation_panel_id value. This should produce the error shown.
Screenshot 2024-07-29 at 4 51 15 PM
Screenshot 2024-07-29 at 4 51 28 PM

This validator also works on existing affiliations if they have a curation_panel_id value, the affiliation type cannot be switched to Independent Curation Panel. The same error is produced when the Save button is selected. Also note that ID's cannot be edited after an affiliation is created.

Screenshot 2024-07-29 at 4 53 34 PM
Screenshot 2024-07-29 at 4 53 46 PM
Screenshot 2024-07-29 at 4 53 55 PM

After selecting GCEP type, providing an invalid curation_panel_id, and selecting "save" the below error is shown.
Screenshot 2024-07-30 at 2 03 30 PM

After selecting VCEP type, providing an invalid curation_panel_id, and selecting "save" the below error is shown.
Screenshot 2024-07-30 at 2 04 14 PM

After providing an invalid affiliation_id and selecting "save" the below error is shown.
Screenshot 2024-07-30 at 2 04 35 PM

### CODE REVIEW

File: src/affiliations/models.py [hunk 1]
```diff
@@ -2,23 +2,49 @@
...
+    """
+    10000 ID. All affiliations will have this ID, however, some affiliations
+    will share this ID. Affiliations that share this ID will have different
+    curation_panel_ids.
+    """
+    affiliation_id: models.IntegerField = models.IntegerField(
+        help_text="10000 number ID"
+    )
+    """
+    4000 or 5000 ID. This ID can be null as independent groups will not have
```
Comment (liammulh) [inline src/affiliations/models.py:21]: Should be 40,000 or 50,000, not 4,000 or 5,000. See https://github.com/ClinGen/gci-vci-aws/blob/dev/gci-vci-serverless/local-files/affiliations.json for examples.

File: src/affiliations/models.py [hunk 2]
```diff
@@ -2,23 +2,49 @@
...
     status: models.CharField = models.CharField()
-    type: models.CharField = models.CharField()
     clinical_domain_working_group: models.CharField = models.CharField()
     members: models.CharField = models.CharField()
 
     def __str__(self):
         """Provide a string representation of an affiliation."""
         return f"Affiliation {self.affiliation_id} {self.full_name}"
 
+    def clean(self):
```
Comment (liammulh) [inline src/affiliations/models.py:39]: Very clean solution! 🧼

**Review by liammulh — APPROVED** [review]
Looking good. Could we also raise a validation error if $10000 \leq id < 20000$ and similarly raise a validation error if you have a curation panel ID that isn't in the correct range (40K range for GCEPs, 50K range for VCEPs)?

File: src/affiliations/models.py (hunk 2 above)
Comment (sanchegm) [reply src/affiliations/models.py:39]: Thanks!

File: src/affiliations/models.py (hunk 1 above)
Comment (sanchegm) [reply src/affiliations/models.py]: Updated! Thank you!

**Review by liammulh — APPROVED** [review]
Looks good!

**Pull request merged**